aws-lc-rs/tests/rsa_test.rs

```diff
@@ -836,10 +873,146 @@ fn max_encrypt_key() {
 
     assert_eq!(message, plaintext);
+
+    let pkcs1_parsed_private =
+        Pkcs1PrivateDecryptingKey::new(parsed_private_key.clone()).expect("supported key");
+    let pkcs1_parsed_public =
+        Pkcs1PublicEncryptingKey::new(parsed_public_key.clone()).expect("supported key");
+
+    let message = vec![42u8; pkcs1_parsed_public.max_plaintext_size()];
+
+    let mut ciphertext = vec![0u8; pkcs1_parsed_public.ciphertext_size()];
+    let ciphertext = pkcs1_parsed_public
+        .encrypt(&message, &mut ciphertext)
+        .expect("encrypted");
+
+    let mut plaintext = vec![0u8; pkcs1_parsed_private.min_output_size()];
+    let plaintext = pkcs1_parsed_private
+        .decrypt(ciphertext, &mut plaintext)
+        .expect("decrypted");
+
+    assert_eq!(message, plaintext);
+}
+
+#[test]
+fn errors_on_larger_than_max_plaintext() {
+    const PUBLIC_KEY: &[u8] = include_bytes!("data/rsa_test_public_key_2048.x509");
+
+    let parsed_public_key = PublicEncryptingKey::from_der(PUBLIC_KEY).expect("key supported");
+
+    let oaep_parsed_public =
+        OaepPublicEncryptingKey::new(parsed_public_key.clone()).expect("supported key");
+
+    let message = vec![42u8; oaep_parsed_public.max_plaintext_size(&OAEP_SHA256_MGF1SHA256) + 1];
+
+    let mut ciphertext = vec![0u8; oaep_parsed_public.ciphertext_size()];
+    oaep_parsed_public
+        .encrypt(&OAEP_SHA256_MGF1SHA256, &message, &mut ciphertext, None)
+        .expect_err("plaintext too large");
+
+    let pkcs1_parsed_public =
+        Pkcs1PublicEncryptingKey::new(parsed_public_key.clone()).expect("supported key");
+
+    let message = vec![42u8; pkcs1_parsed_public.max_plaintext_size() + 1];
+
+    let mut ciphertext = vec![0u8; pkcs1_parsed_public.ciphertext_size()];
+    pkcs1_parsed_public
+        .encrypt(&message, &mut ciphertext)
+        .expect_err("plaintext too large");
 }
 
 #[test]
 fn too_big_encrypt_key() {
     const PRIVATE_KEY: &[u8] = include_bytes!("data/rsa_test_private_key_16384.p8");
     PrivateDecryptingKey::from_pkcs8(PRIVATE_KEY).expect_err("key too big");
 }
+
+macro_rules! round_trip_pkcs1_encryption {
+    ($name:ident, $keysize:expr) => {
+        #[test]
+        fn $name() {
+            const MESSAGE: &[u8] = b"Hello World";
+
+            let priv_key = PrivateDecryptingKey::generate($keysize).expect("key generated");
+            let pub_key = priv_key.public_key();
+
+            let priv_key =
+                Pkcs1PrivateDecryptingKey::new(priv_key).expect("construct PKCS1 private key");
+
+            let pub_key =
+                Pkcs1PublicEncryptingKey::new(pub_key).expect("construct PKCS1 private key");
+
+            let (byte_len, bit_len) = match $keysize {
+                KeySize::Rsa2048 => (256, 2048),
+                KeySize::Rsa3072 => (384, 3072),
+                KeySize::Rsa4096 => (512, 4096),
+                KeySize::Rsa8192 => (1024, 8192),
+                _ => panic!("missing KeySize match arm"),
```

Comment:
Just curious: what does `arm` mean here?

Reply:
The expression on the left of the `=>` is referred to as a "match arm". https://doc.rust-lang.org/reference/expressions/match-expr.html